### PR TITLE
fix: Anoncreds schemas and validation

### PR DESCRIPTION
**Pull request by DaevMithran**

### CODE REVIEW

File: acapy_agent/wallet/did_method.py [hunk 1]
```diff
@@ -97,6 +97,13 @@ def holder_defined_did(self) -> HolderDefinedDid:
     holder_defined_did=HolderDefinedDid.NO,
 )
 
+CHEQD = DIDMethod(
+    name="cheqd",
+    key_types=[ED25519],
+    rotation=True,
+    holder_defined_did=HolderDefinedDid.ALLOWED,
+)
+
```
Comment (dbluhm) [inline acapy_agent/wallet/did_method.py:106]: Unless the Cheqd DID Method will always be available (i.e. without a plugin), we shouldn't put this here. For an example of how to register a DID Method from a plugin, see: https://github.com/Indicio-tech/acapy-did-indy/blob/main/acapy_did_web/__init__.py#L10-L11

File: acapy_agent/wallet/routes.py [hunk 2]
```diff
@@ -312,7 +322,14 @@ class DIDListQueryStringSchema(OpenAPISchema):
...
         validate=validate.OneOf(
-            [KEY.method_name, SOV.method_name, PEER2.method_name, PEER4.method_name]
+            [
+                KEY.method_name,
+                SOV.method_name,
+                TDW.method_name,
+                CHEQD.method_name,
+                PEER2.method_name,
+                PEER4.method_name,
+            ]
```
Comment (dbluhm) [inline acapy_agent/wallet/routes.py:332]: To enable plugged in methods to be queried, we should loosen this validation, probably just removing the `validate.OneOf` altogether.

**Review by dbluhm — COMMENTED** [review]
Overall, looks good! Just a couple of minor points on pluggability of DID Methods and how best to handle that.

File: acapy_agent/wallet/did_method.py (hunk 1 above)
Comment (DaevMithran) [reply acapy_agent/wallet/did_method.py]: @dbluhm thanks for the suggestion. Moved it to plugins!

File: acapy_agent/wallet/routes.py (hunk 2 above)
Comment (DaevMithran) [reply acapy_agent/wallet/routes.py]: Removed it.

**Review by dbluhm — DISMISSED** [review]
LGTM!

**Comment by jamshale** [comment]
Only thing i'm wondering about was why the examples and validation endpoints got changed for /revocation endpoints. These can still only handle indy objects?

**Comment by DaevMithran** [comment]
> Only thing i'm wondering about was why the examples and validation endpoints got changed for /revocation endpoints. These can still only handle indy objects?
> 
> Changing them for /anoncreds/revocation endpoints definitely makes sense.



> Only thing i'm wondering about was why the examples and validation endpoints got changed for /revocation endpoints. These can still only handle indy objects?
> 
> Changing them for /anoncreds/revocation endpoints definitely makes sense.

@jamshale Reverted the /revocation endpoint changes. Thanks!

**Review by jamshale — APPROVED** [review]
👍

**Comment by jamshale** [comment]
Not sure why I can't merge main into this branch with github. When it's up-to-date I'll merge it.

**Comment by jamshale** [comment]
Sorry. Just merged a fix for the coverage reporting. I won't merge anything else if you update this one more time.

**Pull request merged**